proposals/0012-introduce-reactNativeMeta.md

```diff
@@ -0,0 +1,148 @@
+---
+title: Introducing `reactNativeMeta` to `package.json`, for RN specific metadata
+author:
+  - Lorenzo Sciandra
+date: 25-01-2021
+---
+
+# RFC0012: Introducing `reactNativeMeta` to `package.json`, for RN specific metadata
+
+## Summary
+
+This RFC wants to introduce and formally discuss a new section for the package.json of a react-native project (app or library), called `reactNativeMeta`.
```

Comment:
I think the naming will create confusion with our internal stuff so `reactNativeMetadata` would be better.

As React Native is developed by Meta, the `reactNativeMeta` field might give the false impression that this is something related to our internal infrastructure which I believe we should avoid.

Reply:
Just to clarify, this name was chosen because there's prior art: [`peerDependenciesMeta`](https://docs.npmjs.com/cli/v9/configuring-npm/package-json?v=true#peerdependenciesmeta).

> might give the false impression that this is something related to our internal infrastructure

I honestly don't think this will be an issue if the documentation says to use it.

Reply:
> Just to clarify, this name was chosen because there's prior art: [`peerDependenciesMeta`](https://docs.npmjs.com/cli/v9/configuring-npm/package-json?v=true#peerdependenciesmeta).

Thanks for sharing this (I was actually not aware of `peerDependenciesMeta`). I could see `reactNativeMeta` as a valid naming if we were to have a `reactNative` field as well, and this new field provides supporting metadata to the previous configuration. Given that we don't have a `reactNative` config, I'd prefer `reactNativeMetadata` here. I think we have little value in keeping `reactNativeMeta` aligned to `peerDependenciesMeta`.

Reply:
> > Just to clarify, this name was chosen because there's prior art: [`peerDependenciesMeta`](https://docs.npmjs.com/cli/v9/configuring-npm/package-json?v=true#peerdependenciesmeta).
> 
> Thanks for sharing this (I was actually not aware of `peerDependenciesMeta`). I could see `reactNativeMeta` as a valid naming if we were to have a `reactNative` field as well, and this new field provides supporting metadata to the previous configuration. Given that we don't have a `reactNative` config, I'd prefer `reactNativeMetadata` here. I think we have little value in keeping `reactNativeMeta` aligned to `peerDependenciesMeta`.

I'm not particularly attached to the name, and I don't really care what we end up choosing. But I don't quite follow this logic. "Meta" is just a shorter name for "Metadata". If `reactNativeMeta` shouldn't exist because `reactNative` doesn't, then neither should `reactNativeMetadata`.

Reply:
IMO it should just be `reactNative`. `peerDependenciesMeta` kinda makes sense in the world where `peerDependencies` already exists and it well-formed, a new way to extend them was needed. But there are plenty of `package.json` config keys already, and tools like `cosmicconfig` make that pattern common. E.g. `jest: {}` or `beachball: {}` in the package.

Reply:
@NickGerleman we do not want to use `reactNative`, it will lead to confusion as there's already a `react-native` usage in `package.json` because of some web/bundler related stuff, see more https://github.com/nodejs/node/pull/45367